Dynatrace/dynatrace-operator
# Adapt error handling of invalid revision.json

# Description
Currently, if the revision.json file is somehow corrupted, an error is logged in the reconciler, with no hint where that error happened exactly. Additionally if the file exists but is corrupted we always fail with zero chance that it fixes itself, because we only get latest ruxitagenproc from tenant, when the file does not exist. Therefore in case of any error when reading, writing the revision.json file, we should delete it and request newest ruxitagentproc and write it freshly again to the revision.json file

## How can this be tested?
SSH into a cluster where the operator is running, navigate to /var/lib/plugins/csi.oneagent.dynatrace.com/data/<your tenant>
inside that directory there should be a revision.json file -> corrupt it by deleting some characters (like bracket to make it an invalid json) -> wait to see that an precise error is logged in the csi driver and that the file gets deleted and afterwards filled with correct information again 

## Checklist
